File: HW4/HW4Test/ProgramTests.fs
module HW4Tests.ProgramTests

open Xunit
open HW4

[<Fact>]
let ``Main_WrongInputData_WillReturnError`` () =
    let args = [|"332";"/";"?"|]
    let check = Program.main args
    Assert.True(check > 0)

[<Fact>]
let ``Main_CorrectInputData_WillReturnCorrect`` () =
    let args = [|"4";"*";"8"|]
    let check = Program.main args
    let result = check = 0
    Assert.True(result)  

File: HW2/HW2Tests/ProgramTests.cs
using System;
using HW2;
using Xunit;

namespace HW2Tests
{
    public class ProgramTests
    {
        [Fact]
        public void Main_WrongInputData_WillReturnNotZero()
        {
            var args = new[] { "1", "+", "&&&" };
            var result = Program.Main(args);
            Assert.True(result > 0);
        }
        
        [Fact]
        public void Main_CorrectInputData_WillReturnZero()
        {
            var args = new[] { "1", "+", "3" };
            var result = Program.Main(args);
            Assert.True(result == 0);
        }
    }
}